### PR TITLE
Add zeCommandListHostSynchronize

### PR DESCRIPTION
**Pull request by jandres742**
Resolves: #93

### CODE REVIEW

**Comment by jandres742** [comment]
@wdamon-intel , @michalmrozek, @aravindksg, @rdeodhar

please review

File: scripts/core/cmdlist.yml [hunk 1]
```diff
@@ -180,6 +180,33 @@ returns:
...
+            [in] if non-zero, then indicates the maximum time (in nanoseconds) to yield before returning $X_RESULT_SUCCESS or $X_RESULT_NOT_READY;
+            if zero, then immediately returns the status of the immediate command list;
+            if UINT64_MAX, then function will not return until complete or device is lost.
+            Due to external dependencies, timeout may be rounded to the closest value allowed by the accuracy of those dependencies.
+returns:
+    - $X_RESULT_NOT_READY:
+       - "timeout expired"
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+       - "handle does not correspond to an immediate command list"
+--- #--------------------------------------------------------------------------
```
Comment (aravindksg) [inline scripts/core/cmdlist.yml:209]: might be good to match the return codes we have for `CommandQueueSync`?

**Review by rdeodhar — COMMENTED** [review]
LGTM

File: scripts/core/cmdlist.yml (hunk 1 above)
Comment (jandres742) [reply scripts/core/cmdlist.yml:209]: @aravindksg this already has the same codes as queueSync. Only extra is INVALID_ARGUMENT.

**Review by aravindksg — APPROVED** [review]

File: scripts/core/cmdlist.yml [hunk 2]
```diff
@@ -180,6 +180,33 @@ returns:
...
         - "`(nullptr == phWaitEvents) && (0 < numWaitEvents)`"
 --- #--------------------------------------------------------------------------
+type: function
+desc: "Synchronizes an immediate command list by waiting on the host for the completion of all commands previously submitted to it."
+class: $xCommandList
+name: HostSynchronize
+version: "1.6"
+details:
+    - "The application must call this function only with command lists created with $xCommandListCreateImmediate."
+    - "Waiting on one immediate command list shall not block the concurrent execution of commands appended to other immediate command lists."
```
[the remark this answers is not in the document]
Comment (jandres742) [reply scripts/core/cmdlist.yml:190]: Thanks @MichalMrozek . Changed description, please check.

**Review by MichalMrozek — APPROVED** [review]

**Pull request merged**